ansible/configs/osp-stf/software.yml

```diff
@@ -0,0 +1,482 @@
+
+    - name: Wait until Elastic Cloud CSV is installed
+      k8s_info:
+        api_version: operators.coreos.com/v1alpha1
+        kind: ClusterServiceVersion
+        name: "{{ r_subscription_elastic_cloud.resources[0].status.currentCSV }}"
+        namespace: service-telemetry
+      register: r_csv_elastic_cloud
+      retries: 50
+      delay: 20
+      until:
+        - r_csv_elastic_cloud.resources[0].status.phase is defined
+        - r_csv_elastic_cloud.resources[0].status.phase | length > 0
+        - r_csv_elastic_cloud.resources[0].status.phase == "Succeeded"
+
+    - name: Create subscription for Service Telemetry Operator
+      k8s:
+        state: present
+        definition: "{{ lookup('file', 'servicetelemetrysubscription.yaml') }}"
+
+    - name: Get Installed Service Telemetry CSV
+      k8s_info:
+        api_version: operators.coreos.com/v1alpha1
+        kind: Subscription
+        name: servicetelemetry-operator
+        namespace: service-telemetry
+      register: r_subscription_service_telemetry
+      retries: 50
+      delay: 20
+      until:
+        - r_subscription_service_telemetry.resources[0].status.currentCSV is defined
+        - r_subscription_service_telemetry.resources[0].status.currentCSV | length > 0
+
+    - name: Wait until Service Telemetry CSV is installed
+      k8s_info:
+        api_version: operators.coreos.com/v1alpha1
+        kind: ClusterServiceVersion
+        name: "{{ r_subscription_service_telemetry.resources[0].status.currentCSV }}"
+        namespace: service-telemetry
+      register: r_csv_service_telemetry
+      retries: 50
+      delay: 20
+      until:
+        - r_csv_service_telemetry.resources[0].status.phase is defined
+        - r_csv_service_telemetry.resources[0].status.phase | length > 0
+        - r_csv_service_telemetry.resources[0].status.phase == "Succeeded"
+
+    - name: Creating a ServiceTelemetry object in OCP
+      k8s:
+        state: present
+        definition: "{{ lookup('file', 'servicetelemetryobject.yaml') }}"
+
+    - name: Wait until Service Telemetry is ready
+      k8s_info:
+        api_version: infra.watch/v1alpha1
+        kind: ServiceTelemetry
+        name: stf-default
+        namespace: service-telemetry
+      register: servicetelemetry_result
+      retries: 30
+      delay: 20
+      until:
+        - servicetelemetry_result.resources[0].status.conditions[0].status is defined
+        - servicetelemetry_result.resources[0].status.conditions[0].status == "True"
+
+    - name: Project scope to service-telemetry
+      shell: oc project service-telemetry
+
+    - name: sleep for 120 seconds for STF
+      wait_for:
+        timeout: 120
+      delegate_to: localhost
+
+    - name: Get AMQ Interconnect route address
+      shell: oc get routes -ogo-template='{{'{{'}} range .items {{ '}}' }}{{'{{'}}printf "%s\n" .spec.host {{ '}}' }}{{'{{'}} end {{ '}}' }}' | grep "\-5671"
```

Comment:
Thanks, so that you understand I'm following what it's documented here: https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.1/html-single/service_telemetry_framework/index#retrieving-the-qdr-route-address